cedar-policy-core/src/ast/entity.rs

```diff
@@ -203,7 +326,16 @@ impl EntityUID {
 
 impl std::fmt::Display for EntityUID {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}::\"{}\"", self.entity_type(), self.eid.escaped())
+        match self {
+            EntityUID::EntityUID(entity_uid) => write!(
+                f,
+                "{}::\"{}\"",
+                self.entity_type(),
+                entity_uid.eid.escaped()
+            ),
+            #[cfg(feature = "tolerant-ast")]
+            EntityUID::Error => write!(f, "Expr::ExprError"),
```

Comment:
This isn't a valid EntityUID, maybe we should still use `self.entity_type()` like in the other case, and then have something inside quotes so it's a valid EntityUID